.github/workflows/deploy.yaml

```diff
@@ -1,27 +1,81 @@
-# Deploy tagged versions
-name: Deploy
+# Based on https://github.com/pypa/cibuildwheel/blob/main/examples/github-deploy.yml
+name: Build and deploy to PyPI
 
+# Build on every branch push and pull request
+# Also build (and deploy) when a GitHub Release is created
 on:
   push:
-    tags:
-      - '*'
+  pull_request:
+  release:
+    types:
+      - published
 
 jobs:
-  PyPI-sdist:
-    name: Deploy source distribution to PyPI
-    runs-on: ubuntu-latest
+  build_wheels:
+    name: Build wheels on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-20.04, windows-2019, macos-10.15]
+
     steps:
       - uses: actions/checkout@v2
         with:
           fetch-depth: 0  # fetch the entire repo history, required to guarantee versioneer will pick up the tags
+
       - uses: actions/setup-python@v2
+        name: Install Python
         with:
-          python-version: '3.x'
-      - run: pip install twine build
-      - run: python -m build --sdist
-      - run: twine check --strict dist/*
-      - name: "Upload to PyPI"
-        run: twine upload -r pypi dist/pycalphad-*.tar.gz
+          python-version: '3.8'
+
+      - name: Build wheels
+        uses: pypa/cibuildwheel@v2.0.0a2
         env:
-          TWINE_USERNAME: "__token__"
-          TWINE_PASSWORD: ${{ secrets.PYPI_PYCALPHAD_TOKEN }}
\ No newline at end of file
+          # Build all CPython platforms except Python 3.5 and 3.6
+          # PyPy wheels not allowed because SciPy (build requirement) is not available
+          CIBW_BUILD: cp3*-*
+          CIBW_SKIP: cp35-* cp36-*
+          CIBW_ARCHS_MACOS: x86_64 universal2 arm64
+
+      - uses: actions/upload-artifact@v2
+        with:
+          path: ./wheelhouse/*.whl
+
+  build_sdist:
+    name: Build source distribution
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/setup-python@v2
+        name: Install Python
+        with:
+          python-version: '3.8'
+
+      - run: pip install build
+
+      - name: Build sdist
+        run: python -m build --sdist
+
+      - uses: actions/upload-artifact@v2
+        with:
+          path: dist/*.tar.gz
+
+  upload_pypi:
+    needs: [build_wheels, build_sdist]
+    runs-on: ubuntu-latest
+    # upload to PyPI when a GitHub Release is created
+    if: github.event_name == 'release' && github.event.action == 'published'
```

Comment:
So wheels are built on every commit, and only deployed when a GitHub Release is created?

Reply:
Correct